client/dashboard/me/billing-purchases/purchase-settings/purchase-expiring-notice.tsx

```diff
@@ -0,0 +1,308 @@
+							window.location.href = getRenewalUrlFromPurchase( purchase );
+						} }
+					/>
+				) : undefined
+			}
+		>
+			<ExpiringText purchase={ currentPurchase } />
+		</Notice>
+	);
+}
+
+function ExpiringText( { purchase }: { purchase: Purchase } ) {
+	if (
+		purchase.site_slug &&
+		purchase.expiry_status === 'manual-renew' &&
+		purchase.bill_period_days !== SubscriptionBillPeriod.PLAN_CENTENNIAL_PERIOD
+	) {
+		return <ExpiringLaterText purchase={ purchase } />;
+	}
+
+	const purchaseName = purchase.is_domain ? purchase.meta ?? '' : purchase.product_name;
+
+	if ( purchase.bill_period_days === SubscriptionBillPeriod.PLAN_MONTHLY_PERIOD ) {
+		const daysToExpiry = differenceInCalendarDays( new Date( purchase.expiry_date ), new Date() );
+
+		if ( isTemporarySitePurchase( purchase ) ) {
+			return sprintf(
+				// translators: purchaseName is the name of the plan and daysToExpiry is a number of days
+				__( '%(purchaseName)s will expire and be removed in %(daysToExpiry)d days.' ),
+				{
+					purchaseName,
+					daysToExpiry,
+				}
+			);
+		}
+
+		return sprintf(
+			// translators: purchaseName is the name of the plan and daysToExpiry is a number of days
+			__( '%(purchaseName)s will expire and be removed from your site in %(daysToExpiry)d days.' ),
+			{
+				purchaseName,
+				daysToExpiry,
+			}
+		);
+	}
+
+	if ( isTemporarySitePurchase( purchase ) ) {
+		// translators: purchaseName is the name of the plan and expiry is a formatted string like "in 3 months".
+		return sprintf( __( '%(purchaseName)s will expire and be removed %(expiry)s.' ), {
+			purchaseName,
+			expiry: getRelativeTimeString( new Date( purchase.expiry_date ) ),
+		} );
+	}
+
+	return sprintf(
+		// translators: purchaseName is the name of the plan and expiry is a formatted string like "in 3 months".
+		__( '%(purchaseName)s will expire and be removed from your site %(expiry)s.' ),
+		{
+			purchaseName,
+			expiry: getRelativeTimeString( new Date( purchase.expiry_date ) ),
+		}
+	);
+}
+
+export function ExpiringLaterText( {
+	purchase,
+	autoRenewingUpgradesAction,
+}: {
+	purchase: Purchase;
+	autoRenewingUpgradesAction?: () => void;
+} ) {
+	const purchaseName = purchase.is_domain ? purchase.meta ?? '' : purchase.product_name;
+	const expiry = getRelativeTimeString( new Date( purchase.expiry_date ) );
+
+	if ( purchase.payment_type === 'credits' ) {
+		if ( autoRenewingUpgradesAction ) {
+			return createInterpolateElement(
+				sprintf(
+					// translators: purchaseName is the name of the plan and expiry is a formatted string like "in 3 months".
+					__(
+						"You purchased %(purchaseName)s with credits – please update your payment information before your plan expires %(expiry)s so that you don't lose out on your paid features! You also have <link>other upgrades</link> on this site that are scheduled to renew soon."
+					),
+					{ purchaseName, expiry }
+				),
+				{
+					link: <Button onClick={ autoRenewingUpgradesAction } />,
```

Comment:
Should it be `Link`?

Reply:
I don't think so... clicking this text should display the upcoming renewals dialog box, which is definitely a button-like behavior (it doesn't link anywhere). That said, the button should be styled as a link, which is a good point. I'll update these.